orchagent/orchdaemon.cpp

```diff
@@ -313,5 +358,81 @@ void OrchDaemon::start()
         for (Orch *o : m_orchList)
             o->doTask();
 
+        /*
+         * All data to be restored have been added to m_toSync of each orch
+         * at contructor phase.  And the order of m_orchList guranteed the
+         * dependency of tasks had been met, restore is done.
+         */
+        if (!restored && m_select->isQueueEmpty() && gPortsOrch->isInitDone())
+        {
+            /*
+             * drain remaining data that are out of order like LAG_MEMBER_TABLE and VLAN_MEMBER_TABLE
+             * since they were checked before LAG_TABLE and VLAN_TABLE.
+             */
+            for (Orch *o : m_orchList)
+            {
+                o->doTask();
+            }
+
+            warmRestoreValidation();
+            SWSS_LOG_NOTICE("Orchagent state restore done");
+            restored = true;
+            syncd_apply_view();
+
+            // TODO: should be set after port/fdb/arp sync up
+            WarmStart::setWarmStartState("orchagent", WarmStart::RECONCILED);
```

Comment:
how do we sync the port status and fdb notifications?

Reply:
Port state sync: https://github.com/Azure/sonic-swss/pull/557
FDB state sync: https://github.com/Azure/sonic-swss/pull/558
I think @zhenggen-xu  is to propose PR for arp sync up.

Reply:
I feel we should take a different approach to solve port oper status and fdb sync in orchagent. 

Since we have information in asic db, we should just send syncd a command and ask syncd to enumerate all state in asic db and regenerate all the events back to orchagent. 

The ground truth is in asic_db.

Reply:
We may have more in-depth discussion of port state sync up and fdb sync up in the corresponding PRs.  Here is some quick recap:

<1> Current SONiC use NotificationProducer to inform port state change and fdb,  which means orchagent will miss the signal if it happens between orchagent shutdown and startup.

Syncd doens't know whether the signal has been lost or not.  Also that notification event is triggered from libsai/SDK.

<2> In current port sync up implementation,  we do a get SAI_PORT_ATTR_OPER_STATUS via sai API to have the latest oper status, then update hostif and db accordingly.  

<3> For FDB sync up,   since syncd does put the latest FDB entry into ASIC DB,  yes, we are retrieving the latest FDB from ASIC.

Surely there is some other alternative, like change the Notification mechanism to some db data persistent channel. Under current framework, the above approaches have proven to be working and stable.  I should have put more comments to the port state and fdb sync up implementation.

syncd_apply_view() is called after state restore,  I guess that could make the life of syncd view comparison a little easier, but it is up to how you want view comparison to work.  The libsai redis idempotent API solution doesn't have dependency on  syncd_apply_view().

Reply:
we can ask syncd to pause sending notification when we are upgrading.

it is better to ask syncd to re-send the previous view to us. we do not orchagent to scan the asic view because the format might change and it can break orchagent code. orchagent should not look at the asic db.

Can you modify you fdb sync and port sync to syncd?

Reply:
For port state sync, I don't think there is need to make the solution so complex.
FDB format could be a valid concern, but we should alway be careful when making format change.

My understanding of SONiC design about docker is to decouple the services and make them more independent from each other. But I'm seeing the intent to couple swss and syncd tightly together, what is the real purpose behind this?  Why were syncd and orchagent put into different dockers at the first place?